manual/src/cmds/profile-perf.etex

```diff
@@ -0,0 +1,90 @@
+        perf record -F 99 --call-graph fp <YOUR_EXECUTABLE>
+\end{verbatim}
+
+The \texttt{-F 99} option tells \texttt{perf} to sample at 99Hz, which avoids generating excessive data for longer runs and is unlikely to be in lockstep with other periodic activities. The \texttt{--call-graph fp} instructs \texttt{perf} to use frame pointers to get the call-graph, followed by the OCaml executable you want to profile. This command creates a \texttt{perf.data} file in the current directory. Alternatively, use \texttt{--output} to choose a more descriptive filename.
+
+The \texttt{perf record} command works by copying a segment of the call stack at each sample, recording this data into a \texttt{perf.data} file. These samples can then be processed after recording using \texttt{perf report} to reconstruct the profiled program’s call stack at every sample.
+
+\texttt{perf record --call-graph} has multiple options available for recording call graphs:
+\begin{itemize}
+\item Frame Pointers (default)
+\item DWARF's Call Frame Information (CFI)
+\item Hardware Last Branch Record (LBR) available on certain Intel CPUs
+\end{itemize}
+
+\texttt{perf} uses the symbols in an OCaml executable, so it helps to understand OCaml's name-mangling scheme to map names to OCaml source locations. Before OCaml 5.1, \texttt{ocamlopt} mangled names using the \texttt{camlModule__identifier_stamp} format; from 5.1 onwards, the separator is a dot: \texttt{camlModule.identifier_stamp}. Both formats are supported by \texttt{perf}.
+
+Consider the following program:
+\begin{caml_example*}{verbatim}
+module Compute = struct
+  let rec fib n = 
+    if n = 0 then 0
+    else if n = 1 then 1
+    else fib (n-1) + fib (n-2)
+end
+
+let main () = 
+  let r = Compute.fib 20 in 
+  Printf.printf "fib(20) = %d" r
+
+let _ = main ()
+\end{caml_example*}
+
+This program produces the names \texttt{camlFib.main_274} for the \texttt{main} function and \texttt{camlFib.fib_271} for the \texttt{fib} function in the \texttt{Compute} module.
+
+\section{s:ocamlperf-printing}{Printing profiling information}
+
+The \texttt{perf report} command summarises the captured \texttt{perf.data} file. The basic \texttt{perf report} command is:
+
+\begin{verbatim}
+        perf report -f --no-children perf.data
+\end{verbatim}
+
+This displays the accumulated call-graphs in an interactive terminal interface where you can navigate the data selecting functions and threads for more information. Alternatively \texttt{--stdio} will output similar data using a text based report writing to \texttt{stdout}.
+
+Profile data can be turned into Flame Graphs, a visualisation of hierarchical data, created to visualise stack traces of profiled software so that the most frequent code-paths can be identified quickly and accurately. Use the scripts \texttt{stackcollapse-perf.pl} and \texttt{flamegraph.pl} found at Brendan Gregg's \href{https://www.brendangregg.com/flamegraphs.html}{Flame Graphs} web page as follows:
```

Comment:
```suggestion
Profile data can be visualised as Flame Graphs, which highlight the most frequent code paths in stack traces. Use the scripts \texttt{stackcollapse-perf.pl} and \texttt{flamegraph.pl} found at Brendan Gregg's \href{https://www.brendangregg.com/flamegraphs.html}{Flame Graphs} web page as follows:
```

I know this suggested first sentence is way more concise, so I want to ensure it's still accurate. Is it necessary to define what a Flame Graph is to your audience? If so, perhaps: 

Profile data can be turned into Flame Graphs, a visualisation of hierarchical data, to highlight the most frequent code-paths in stack traces.

If necessary, add the "of profiled software" to the end of the sentence.